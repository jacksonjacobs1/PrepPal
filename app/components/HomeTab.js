import React, { useState } from 'react';
import { getAuth } from '@firebase/auth';
import { Button, DefaultTheme} from 'react-native-paper';
import { StyleSheet, Text, View, List } from 'react-native';

export default function HomeTab({ navigation }) {
  const auth = getAuth();
  const user = auth.currentUser;

  const name = user.email
  const [recipe, setFood] = useState([
      {recipe: 'Recipe 1', key: '1'},
      {recipe: 'Recipe 2', key: '2'},
      {recipe: 'Recipe 3', key: '3'},
      {recipe: 'Recipe 4', key: '4'},
      {recipe: 'Recipe 5', key: '5'}
    ]);
  return (
    <View style={styles.container}>
    <Text style={styles.text1} > Hello {name}, Welcome to PrepPal! </Text>
    <Text style={styles.listItem}>Recent Recipes:</Text>
    { recipe.map((item) => {
      return(
        <View key = {item.key}>
        <Text style={styles.listItem}>{item.recipe}</Text>
        </View>
      )
    })}
    <Button mode="contained" theme={DefaultTheme} testID="WORK" style={styles.button} onPress={() => navigation.navigate('Saved Recipes')}>
      See All Saved Recipes
    </Button>
    </View>
  );
}


const theme = {
  ...DefaultTheme,
  roundness: 3,
  colors: {
    ...DefaultTheme.colors,
    background: 'white',
  },
};

const styles = StyleSheet.create({
  container: {
    marginTop: 0,
    flex: 1
  },
  text1: {
    color: '#6200ee',
    fontWeight: 'bold',
    fontSize: 30,
    borderWidth: 1,
    textAlign: 'center',
    backgroundColor:'white',
    fontWeight:'bold'
  },
  listItem:{
    marginTop:5,
    padding:20,
    backgroundColor: '#f6f6f6',
    fontSize: 20,
    textAlign: 'center',
  },
  button:{
    marginTop:65,
    justifyContent:'flex-start',
  }
});