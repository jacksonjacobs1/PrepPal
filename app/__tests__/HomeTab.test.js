import React from "react";
import renderer from 'react-test-renderer';
import HomeTab from '../components/HomeTab';
import { cleanup, render } from '@testing-library/react-native';

afterEach(cleanup);

describe('HomeTab', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<HomeTab />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('displays a search bar', () => {
    const { getByTestId } = render(<HomeTab />);
    expect(getByTestId('savedSearch')).toBeTruthy();
  });
});