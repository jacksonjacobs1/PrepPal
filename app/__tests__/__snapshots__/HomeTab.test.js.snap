// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeTab renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#0CC3F9",
      "flex": 1,
      "justifyContent": "flex-start",
    }
  }
  testID="homeTab"
>
  <Text
    style={
      Object {
        "backgroundColor": "yellow",
        "borderColor": "#20232a",
        "borderRadius": 6,
        "borderWidth": 3,
        "color": "black",
        "fontSize": 30,
        "fontWeight": "bold",
        "marginTop": 0,
        "paddingVertical": 10,
        "textAlign": "center",
      }
    }
  >
    Hello 
    Richard
    , Welcome to PrepPal! 
  </Text>
  <Text
    style={
      Object {
        "backgroundColor": "blue",
        "borderColor": "#20232a",
        "borderRadius": 6,
        "borderWidth": 3,
        "color": "red",
        "fontSize": 30,
        "fontWeight": "bold",
        "marginTop": 0,
        "paddingVertical": 6,
        "textAlign": "center",
      }
    }
  >
    Recent Recipes:
    

    chicken flambae
    

    eggs benedict
    

    chocolate chip cookies
     
  </Text>
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
            Object {
              "color": "#E5101C",
            },
          ]
        }
      >
        See All Saved Recipes
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#0CC3F9",
      "flex": 1,
      "justifyContent": "flex-start",
    }
  }
  testID="homeTab"
>
  <Text
    style={
      Object {
        "backgroundColor": "yellow",
        "borderColor": "#20232a",
        "borderRadius": 6,
        "borderWidth": 3,
        "color": "black",
        "fontSize": 30,
        "fontWeight": "bold",
        "marginTop": 0,
        "paddingVertical": 10,
        "textAlign": "center",
      }
    }
  >
    Hello 
    Richard
    , Welcome to PrepPal! 
  </Text>
  <Text
    style={
      Object {
        "backgroundColor": "blue",
        "borderColor": "#20232a",
        "borderRadius": 6,
        "borderWidth": 3,
        "color": "red",
        "fontSize": 30,
        "fontWeight": "bold",
        "marginTop": 0,
        "paddingVertical": 6,
        "textAlign": "center",
      }
    }
  >
    Recent Recipes:
    

    chicken flambae
    

    eggs benedict
    

    chocolate chip cookies
     
  </Text>
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
            Object {
              "color": "#E5101C",
            },
          ]
        }
      >
        See All Saved Recipes
      </Text>
    </View>
  </View>
</View>
`;
